src/Http/Owin/src/OwinEnvironment.cs

```diff
@@ -478,4 +399,210 @@ public FeatureMap(Func<TFeature, object> getter, Func<object> defaultFactory, Ac
+
+            { OwinConstants.SendFiles.SendAsync, new FeatureMap<IHttpResponseBodyFeature>(feature => new SendFileFunc(feature.SendFileAsync)) },
+            { OwinConstants.Security.User, new FeatureMap<IHttpAuthenticationFeature>(feature => feature.User, ()=> null, (feature, value) => feature.User = Utilities.MakeClaimsPrincipal((IPrincipal)value), () => new HttpAuthenticationFeature()) },
+            { OwinConstants.RequestId, new FeatureMap<IHttpRequestIdentifierFeature>(feature => feature.TraceIdentifier, ()=> null, (feature, value) => feature.TraceIdentifier = (string)value, () => new HttpRequestIdentifierFeature()) },
+            { OwinConstants.CallCancelled, new FeatureMap<IHttpRequestLifetimeFeature>(feature => feature.RequestAborted) },
+
+            { OwinConstants.CommonKeys.ClientCertificate, new FeatureMap<ITlsConnectionFeature>(feature => feature.ClientCertificate, (feature, value) => feature.ClientCertificate = (X509Certificate2)value) },
+            { OwinConstants.CommonKeys.LoadClientCertAsync, new FeatureMap<ITlsConnectionFeature>(feature => new Func<Task>(() => feature.GetClientCertificateAsync(CancellationToken.None))) },
+            { OwinConstants.WebSocket.AcceptAlt, new FeatureMap<IHttpWebSocketFeature>(feature => new WebSocketAcceptAlt(feature.AcceptAsync)) }
+        };
+
+        /// <remarks>
+        /// Will be used, only if <see cref="FeatureMaps"/> or <see cref="Clear"/> is called from user-code.
+        /// Is a deep-copy of the singleton <see cref="_entries"/>
+        /// </remarks>
+        private IDictionary<string, FeatureMap> _contextEntries;
+
+        /// <remarks>
+        /// In order not to copy the whole dictionary of featureMaps per request,
+        /// and since OWIN allows the <see cref="Remove(string)"/> operation,
+        /// it's more lightweight to keep track of deleted keys.
+        /// </remarks>
+        private HashSet<string> _deletedKeys;
+
+        /// <remarks>
+        /// There are some <see cref="FeatureMap"/> entries that are context-dependent.
+        /// This dictionary is allocated per request, but does not contain as many entries as <see cref="_entries"/>.
+        /// </remarks>
+        private readonly IDictionary<string, FeatureMap> _contextDependentEntries;
+
+        public OwinEntries(HttpContext context)
+        {
+            _contextDependentEntries = new Dictionary<string, FeatureMap>
+            {
+                { OwinConstants.ResponseBody, new FeatureMap<IHttpResponseBodyFeature>(feature => feature.Stream, () => Stream.Null, (feature, value) => context.Response.Body = (Stream)value) }, // DefaultHttpResponse.Body.Set has built in logic to handle replacing the feature.
+            };
+        }
+
+        static string PortToString(int port) => port switch
+        {
+            80 => "80",
+            443 => "443",
+            8080 => "8080",
```

Comment:
`8080` and `8081` seem a little arbitrary for the more general case the library is dealing with.

Reply:
removed 8080 and 8081 -> left only 80 and 443